### PR TITLE
WOE 20 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/galvanic_giant_storm_reading.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Name:Galvanic Giant
+ManaCost:3 U
+Types:Creature Giant Wizard
+PT:3/3
+T:Mode$ SpellCast | ValidCard$ Creature.cmcGE5 | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigTap | TriggerDescription$ Whenever you cast a spell with mana value 5 or greater, tap target creature an opponent controls and put a stun counter on it.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/galvanic_giant_storm_reading.txt:5]: wrong Valid

File: forge-gui/res/cardsfolder/upcoming/grabby_giant_thats_mine.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Name:Grabby Giant
+ManaCost:3 R
+Types:Creature Giant
+PT:4/3
+K:Reach
+A:AB$ Draw | Cost$ T Sac<1/Artifact;Land/artifact or land> | SpellDescription$ Draw a card.
+DeckHas:Ability$Token|Sacrifice & Type$Treasure|Artifact
+DeckHints:Type$Artifact
+AlternateMode:Adventure
+Oracle:Haste\n{2}{R}, Sacrifice an artifact or land: Draw a card.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/grabby_giant_thats_mine.txt:10]: which KW?

File: forge-gui/res/cardsfolder/upcoming/hamlet_glutton.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Name:Hamlet Glutton
+ManaCost:5 G G
+Types:Creature Giant
+PT:6/6
+K:Bargain
+K:Trample
+S:Mode$ ReduceCost | ValidCard$ Card.Self | Type$ Spell | Amount$ X | EffectZone$ All | Description$This spell costs {2} less to cast if it's bargained.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/hamlet_glutton.txt:7]: check with ValidSpell$ and hardcode Amount

File: forge-gui/res/cardsfolder/upcoming/horned_loch_whale_lagoon_breach.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
+Name:Horned Loch-Whale
+ManaCost:4 U U
+Types:Creature Whale
+PT:6/6
+K:Flash
+K:Ward:2
+K:ETBReplacement:Other:LandTapped
+SVar:LandTapped:DB$ Tap | Defined$ Self | ETB$ True | ConditionOpponentTurn$ True | SpellDescription$ CARDNAME enters the battlefield tapped unless it's your turn.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/horned_loch_whale_lagoon_breach.txt:8]: probably need some PlayerCount check for your ally turns

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/horned_loch_whale_lagoon_breach.txt:8]: I  probably would add a more generic Turn Check for this

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/horned_loch_whale_lagoon_breach.txt:8]: @Simisays do this: `ConditionPlayerTurn$ False`
https://github.com/Card-Forge/forge/blob/master/forge-game/src/main/java/forge/game/spellability/SpellAbilityCondition.java#L335

File: forge-gui/res/cardsfolder/upcoming/hamlet_glutton.txt (hunk 3 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/hamlet_glutton.txt]: current version not working

File: forge-gui/res/cardsfolder/upcoming/horned_loch_whale_lagoon_breach.txt (hunk 4 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/horned_loch_whale_lagoon_breach.txt]: thanks, working now

File: forge-gui/res/cardsfolder/upcoming/hamlet_glutton.txt (hunk 3 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/hamlet_glutton.txt]: Try `Spell.Bargained`

File: forge-gui/res/cardsfolder/upcoming/hearth_elemental_stoke_genius.txt [hunk 5]
```diff
@@ -0,0 +1,19 @@
+Name:Hearth Elemental
+ManaCost:5 R
+Types:Creature Elemental
+PT:4/5
+S:Mode$ ReduceCost | ValidCard$ Card.Self | Type$ Spell | Amount$ X | EffectZone$ All | Description$ This spell costs {X} less to cast, where X is the number of cards in your graveyard that are instant cards, sorcery cards, and/or have an Adventure.
+SVar:X:Count$ValidGraveyard Instant.YouOwn,Sorcery.YouOwn,AdventureCard.YouOwn
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/hearth_elemental_stoke_genius.txt:6]: "AdventureCard" is not a valid Card type

File: forge-gui/res/cardsfolder/upcoming/high_fae_negotiator.txt [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+ManaCost:3 B B
+Types:Creature Faerie Warlock
+PT:3/5
+K:Bargain
+K:Flying
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self+bargained | Execute$ TrigLoseLife | TriggerDescription$ When CARDNAME enters the battlefield, if it was bargained, each opponent loses 3 life and you gain 3 life.
+SVar:TrigLoseLife:DB$ LoseLife | LifeAmount$ 3 | Defined$ Opponent | SubAbility$ DBGainLife
+SVar:DBGainLife:DB$ GainLife | LifeAmount$ 3
+DeckHas:Ability$Sacrifice|LifeGain
+DeckHints:Type$Token|Enchantment|Artifact
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/high_fae_negotiator.txt:11]: Token is not a valid type

File: forge-gui/res/cardsfolder/upcoming/hamlet_glutton.txt (hunk 3 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/hamlet_glutton.txt]: thanks seems to work

**Review by tool4ever — APPROVED** [review]

**Pull request merged**